File: app11_swift_ios/SwiftGuardApp/Resources/CodeSamples/scala/broken-access-control-check_defense.scala
// SECURE pattern
case GET -> Root / "invoices" / LongVar(id) as user =>
  // WHY: findByIdAndOwner scopes the row lookup to the authenticated user at the query level
  invoiceRepo.findByIdAndOwner(id, user.id).flatMap {
    case Some(invoice) => Ok(invoice)
    case None          => Forbidden()
  }

File: app11_swift_ios/SwiftGuardApp/Resources/CodeSamples/go/broken-access-control-check_defense.go
// SECURE pattern
func getInvoice(w http.ResponseWriter, r *http.Request) {
    id := chi.URLParam(r, "id")
    userID := auth.UserIDFromContext(r.Context())
    // WHY: the repository query itself scopes by owner, so a mismatched id returns "not found", not someone else's data
    invoice, err := repo.FindByIDAndOwner(r.Context(), id, userID)
    if err != nil {
        http.Error(w, "not found", http.StatusNotFound)
        return
    }
    json.NewEncoder(w).Encode(invoice)
}
